parkbenchcommon/broadcaster.py

```diff
@@ -0,0 +1,138 @@
+        method creates any necessary spool directories, sets their permissions, and mounts a
+        ramdisk if necessary.
+
+        program_name: The name of the program issuing the broadcast.
+        broadcast_name: The name of the broadcast to issue.
+        uid: The UID of the calling program.
+        gid: The GID of the calling program.
+        """
+
+        self.logger = logging.getLogger(__name__)
+
+        self.logger.debug("Initializing broadcaster for broadcast %s from program %s.",
+                          broadcast_name, program_name)
+
+
+        broadcast_path = '/var/spool/'
+        self.broadcast_path = os.path.join(broadcast_path, program_name, broadcast_name)
+        self.program_name = program_name
+        self.broadcast_name = broadcast_name
+
+        tmpfs_path = os.path.join(broadcast_path, program_name)
+        tmpfs.mount_tmpfs(tmpfs_path, TMPFS_SIZE)
+
+        self._create_directory(self.broadcast_path)
+        # Set permissions to xrw-------
+        group_rw_mode = stat.S_IXUSR | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXGRP | stat.S_IRGRP
+        self._set_file_permissions(self.broadcast_path, group_rw_mode, uid, gid)
+
+        self.logger.info("Broadcaster for eacon %s from program %s initialized.",
+                         broadcast_name, program_name)
+
+    def issue(self):
+        """Place a new file in the broadcast directory. Will raise an exception if it fails."""
+        self.logger.info(
+            "Sending broadcast %s for program %s.", self.broadcast_name, self.program_name)
+        now = datetime.datetime.now().isoformat()
+        random_number = os.urandom(16).encode('hex')
+
+        broadcast_filename = '%s---%s' % (now, random_number)
+        final_path = os.path.join(self.broadcast_path, broadcast_filename)
+
+        try:
+            previous_broadcasts = os.listdir(self.broadcast_path)
+            open(final_path, 'a').close()
+
+            for broadcast_file in previous_broadcasts:
+                os.remove(os.path.join(self.broadcast_path, broadcast_file))
+
+        except Exception as exception:
+            message = \
+                'Could not create broadcast file for broadcast %s, program %s. %s:%s\n%s' \
+                % (self.broadcast_name, self.program_name, type(exception).__name__,
+                   str(exception), traceback.format_exc())
+            self.logger.critical(message)
+            # TODO: Implement exception chaining when we move to Python 3.
+            raise BroadcasterBroadcastException(message)
+
+    def _set_file_permissions(self, path, mode, uid, gid):
+        """Sets permissions for a file. Will raise an exception if it fails.
+
+        path: The file to modify.
+        mode: The mode to set.
+        uid: The owner to set.
+        gid: The group to set.
+        """
+        try:
+            os.chown(path, uid, gid)
+            os.chmod(path, mode)
```

Comment:
These two commands do not set the permissions on parent directories. Permissions should be set on all created directories under (but not including) /var/spool. See what I did in proper-daemonization.